src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -1367,6 +1369,44 @@ def create_rotation_matrices(self):
                                              numpy.radians(float(self.txtSampleRoll.text()))])
         return env, sample*env
 
+    def check_large_calculation(self, num_points):
+        """
+        Generate a popup warning the user about a large calculation
+        :return True if the calculation should continue, False otherwise.
+        """
+
+        if self.warned_user_large_calc or num_points < 50_000: return True
+        # timings for an i7-1165G7 4-core CPU from 2020:
+        # 50k: 1.8s | 100k: 6.3s | 200k: 28.7s | 300k: 65.6s | 400k: 117.2s | 500k: 181.2s
+        # multiply by 2 & round to get a conservative estimate
```

Comment:
It may be good to keep this numbers as a configurable variable. It is now being tuned for the current calculator but we may potentially set different limits if in the future we for example change approach